### PR TITLE
ft_printf: update undefined behavior

### PR DESCRIPTION
**Pull request by pouet**

### CODE REVIEW

File: includes/ft_printf_list.sh [hunk 1]
```diff
@@ -15,17 +15,17 @@ then
...
+	CHK_FT_PRINTF_LIST2+=("0" "% h")
+	CHK_FT_PRINTF_LIST2+=("0" "%Z")
+	CHK_FT_PRINTF_LIST2+=("0" "% hZ")
 
 	# %
 	CHK_FT_PRINTF_LIST+=("0" "%%")
 	CHK_FT_PRINTF_LIST+=("0" "%5%")
 	CHK_FT_PRINTF_LIST+=("0" "%-5%")
-	CHK_FT_PRINTF_LIST+=("0" "%05%")
+	CHK_FT_PRINTF_LIST2+=("0" "%05%")
```
Comment (jgigault) [inline includes/ft_printf_list.sh]: Why should it be undefined behavior?
It does mean "at least 5 characters to print, right-aligned, filled with '0' on right-side", isn't it?

Comment (jgigault) [inline includes/ft_printf_list.sh]: Filled with on left-side**

File: includes/ft_printf_list.sh [hunk 2]
```diff
@@ -34,7 +34,7 @@ then
 	CHK_FT_PRINTF_LIST2+=("s" "% Z |test")
 	CHK_FT_PRINTF_LIST2+=("s" "% Z%s|test")
 	CHK_FT_PRINTF_LIST+=("s" "%%|test")
-	CHK_FT_PRINTF_LIST2+=("s" "%   %|test")
+	CHK_FT_PRINTF_LIST+=("s" "%   %|test")
```
Comment (jgigault) [inline includes/ft_printf_list.sh]: This one is a "bad" use of printf: One space is enough, two or more spaces doesn't mean anything.

File: includes/ft_printf_list.sh [hunk 3]
```diff
@@ -99,7 +99,7 @@ then
 	CHK_FT_PRINTF_LIST+=("s" "Line Feed %s|\n")
 
 	CHK_FT_PRINTF_LIST+=("s" "%10s is a string|this")
-	CHK_FT_PRINTF_LIST+=("s" "%010s is a string|this")
+	CHK_FT_PRINTF_LIST2+=("s" "%010s is a string|this")
```
Comment (jgigault) [inline includes/ft_printf_list.sh]: As the first commentary: 010 means "at least 10 characters to print, right-aligned, filled with '0' on left-side"

File: includes/ft_printf_list.sh [hunk 4]
```diff
@@ -127,7 +127,7 @@ then
 	# c
 	CHK_FT_PRINTF_LIST+=("dc" "%c|42")
 	CHK_FT_PRINTF_LIST+=("dc" "%5c|42")
-	CHK_FT_PRINTF_LIST+=("dc" "%05c|42")
+	CHK_FT_PRINTF_LIST2+=("dc" "%05c|42")
```
Comment (jgigault) [inline includes/ft_printf_list.sh]: Idem

File: includes/ft_printf_list.sh [hunk 5]
```diff
@@ -175,28 +175,28 @@ then
...
+	CHK_FT_PRINTF_LIST+=("d" "%+ d|42")
+	CHK_FT_PRINTF_LIST+=("d" "%+ d|-42")
+	CHK_FT_PRINTF_LIST+=("d" "%  +d|42")
+	CHK_FT_PRINTF_LIST+=("d" "%  +d|-42")
+	CHK_FT_PRINTF_LIST+=("d" "%+  d|42")
+	CHK_FT_PRINTF_LIST+=("d" "%+  d|-42")
+	CHK_FT_PRINTF_LIST+=("d" "% ++d|42")
+	CHK_FT_PRINTF_LIST+=("d" "% ++d|-42")
+	CHK_FT_PRINTF_LIST+=("d" "%++ d|42")
+	CHK_FT_PRINTF_LIST+=("d" "%++ d|-42")
```
Comment (jgigault) [inline includes/ft_printf_list.sh]: A 'space' means "print the sign of the number for negative values only".
A '+' means "print the sign of the number for both positive and negative values".

Isn't it inconsistent?

File: includes/ft_printf_list.sh (hunk 1 above)
Comment (pouet) [inline includes/ft_printf_list.sh]: > For d, i, o, u, x, X, a, A, e, E, f, F, g, and G conversions,[...] for other conversions, the behavior is undefined.
:)

File: includes/ft_printf_list.sh (hunk 2 above)
Comment (pouet) [inline includes/ft_printf_list.sh]: It doesn't mean anything, but not wrong... ^^

File: includes/ft_printf_list.sh (hunk 3 above)
Comment (pouet) [inline includes/ft_printf_list.sh]: Look my first com. 0 flag is only for numbers, others are undefined behavior.

File: includes/ft_printf_list.sh (hunk 4 above)
Comment (pouet) [inline includes/ft_printf_list.sh]: Idem :)

File: includes/ft_printf_list.sh (hunk 5 above)
Comment (pouet) [inline includes/ft_printf_list.sh]: Norme say :
> If the space and + flags both appear, the space flag is ignored.

**Comment by jgigault** [comment]
OK you seem to be pretty well documented :)
Thank You!

**Comment by pouet** [comment]
You're welcome. :)

**Pull request merged**